HDF5 build fails using `-DHDF5_ENABLE_PARALLEL=ON`

Hi devs, 
I'm trying to build the latest develop branch of HDF5, using the below CMake command:
```
cmake .. -DCMAKE_INSTALL_PREFIX=/opt/softeware/hdf5-gnu -DHDF5_BUILD_FORTRAN=ON -DHDF5_ENABLE_ZLIB_SUPPORT=ON -DHDF5_ENABLE_SZIP_SUPPORT=ON -DHDF5_ENABLE_PARALLEL=ON
```
and then execute `make install -j24` to build and install HDF5. Unfortunately, it fails with error: 
```
[ 99%] Linking Fortran executable ../../../bin/H5_f90_h5ex_g_visit
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_dataset.F90:4:10:

    4 |      USE MPI
      |          1
Fatal Error: Cannot open module file ‘mpi.mod’ for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/build.make:75: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/ph5_f90_dataset.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16396: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 99%] Building Fortran object fortran/test/CMakeFiles/testhdf5_fortran.dir/fortranlib_test.F90.o
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_file_create.F90:8:10:

    8 |      USE MPI
      |          1
Fatal Error: Cannot open module file ‘mpi.mod’ for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_file_create.dir/build.make:75: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_file_create.dir/ph5_f90_file_create.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16427: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_file_create.dir/all] Error 2
[ 99%] Building Fortran object HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/ph5_f90_hyperslab_by_chunk.F90.o
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_col.F90:7:10:

    7 |      USE mpi
      |          1
Fatal Error: Cannot open module file ‘mpi.mod’ for reading at (1): No such file or directory
compilation terminated.
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_pattern.F90:8:10:

    8 |      USE MPI
      |          1
Fatal Error: Cannot open module file ‘mpi.mod’ for reading at (1): No such file or directory
compilation terminated.
[ 99%] Linking Fortran executable ../../../bin/H5_f90_h5ex_t_regref_F03
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/build.make:75: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/ph5_f90_hyperslab_by_col.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16489: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/all] Error 2
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/build.make:75: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/ph5_f90_hyperslab_by_pattern.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16520: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/all] Error 2
[ 99%] Built target H5_f90_h5ex_g_iterate
[ 99%] Built target H5_f90_h5ex_t_objrefatt_F03
[ 99%] Built target H5_f90_h5ex_t_opaqueatt_F03
[ 99%] Linking Fortran executable ../../../bin/H5_f90_h5ex_t_string_F03
[ 99%] Linking Fortran executable ../../../bin/H5_f90_h5ex_t_stringC_F03
[ 99%] Linking Fortran executable ../../../bin/H5_f90_h5ex_t_vlenatt_F03
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_row.F90:7:10:

    7 |      USE MPI
      |          1
Fatal Error: Cannot open module file ‘mpi.mod’ for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/build.make:75: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/ph5_f90_hyperslab_by_row.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16458: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/all] Error 2
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_chunk.F90:7:10:

    7 |      USE MPI
      |          1
Fatal Error: Cannot open module file ‘mpi.mod’ for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/build.make:75: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/ph5_f90_hyperslab_by_chunk.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16551: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/all] Error 2
```
I have no idea why it fails with not finding "mpi.mod". The MPI program I use is OpenMPI 5.0.9, and its "mpi.mod" file is in `/opt/software/openmpi/lib`, together with `libmpi_usempif08.so` which is found in configuration step.

Here are my log files of configuration and building step:
[configure.log](https://github.com/user-attachments/files/25120421/configure.log)
[make.log](https://github.com/user-attachments/files/25120430/make.log)

While I try copying those *.mod file from `/opt/software/openmpi/lib` to `/opt/software/openmpi/include` and recompile HDF5, it fails with below message:
```
[ 99%] Built target H5_f90_h5ex_g_visit
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/ph5_f90_hyperslab_by_pattern.F90.o: undefined reference to symbol 'mpi_init_'
/usr/bin/ld: /opt/software/openmpi/lib/libmpi_mpifh.so.40: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/ph5_f90_hyperslab_by_col.F90.o: undefined reference to symbol 'mpi_init_'
/usr/bin/ld: /opt/software/openmpi/lib/libmpi_mpifh.so.40: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/build.make:104: bin/H5_f90_ph5_f90_hyperslab_by_pattern] Error 1
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/build.make:104: bin/H5_f90_ph5_f90_hyperslab_by_col] Error 1
make[1]: *** [CMakeFiles/Makefile2:16520: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:16489: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/all] Error 2
[ 99%] Built target H5_f90_h5ex_g_traverse
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_hyperslab_by_row
[ 99%] Built target H5_f90_exlite
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_file_create
[ 99%] Built target H5_f90_h5ex_t_vlenatt_F03
[ 99%] Built target H5_f90_h5ex_t_vlstring
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/ph5_f90_hyperslab_by_row.F90.o: undefined reference to symbol 'mpi_init_'
/usr/bin/ld: /opt/software/openmpi/lib/libmpi_mpifh.so.40: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/build.make:104: bin/H5_f90_ph5_f90_hyperslab_by_row] Error 1
[ 99%] Built target H5_f90_h5ex_t_vlen_F03
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_dataset
make[1]: *** [CMakeFiles/Makefile2:16458: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/all] Error 2
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_hyperslab_by_chunk
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_file_create.dir/ph5_f90_file_create.F90.o: undefined reference to symbol 'mpi_init_'
/usr/bin/ld: /opt/software/openmpi/lib/libmpi_mpifh.so.40: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_file_create.dir/build.make:104: bin/H5_f90_ph5_f90_file_create] Error 1
make[1]: *** [CMakeFiles/Makefile2:16427: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_file_create.dir/all] Error 2
[ 99%] Linking C executable ../../bin/h5gentest
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_dataset.dir/ph5_f90_dataset.F90.o: undefined reference to symbol 'mpi_init_'
/usr/bin/ld: /opt/software/openmpi/lib/libmpi_mpifh.so.40: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/ph5_f90_hyperslab_by_chunk.F90.o: undefined reference to symbol 'mpi_init_'
/usr/bin/ld: /opt/software/openmpi/lib/libmpi_mpifh.so.40: error adding symbols: DSO missing from command line
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/build.make:104: bin/H5_f90_ph5_f90_dataset] Error 1
make[1]: *** [CMakeFiles/Makefile2:16396: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/all] Error 2
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/build.make:104: bin/H5_f90_ph5_f90_hyperslab_by_chunk] Error 1
make[1]: *** [CMakeFiles/Makefile2:16551: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/all] Error 2
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ex_ds1
[ 99%] Built target h5gentest
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_filtered_writes_no_sel
[ 99%] Built target H5_f90_ex_ds1
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_filtered_writes_no_sel.dir/ph5_f90_filtered_writes_no_sel.F90.o: undefined reference to symbol 'mpi_init_'
/usr/bin/ld: /opt/software/openmpi/lib/libmpi_mpifh.so.40: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_filtered_writes_no_sel.dir/build.make:104: bin/H5_f90_ph5_f90_filtered_writes_no_sel] Error 1
make[1]: *** [CMakeFiles/Makefile2:16582: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_filtered_writes_no_sel.dir/all] Error 2
[ 99%] Linking C executable ../bin/t_filters_parallel
[ 99%] Built target t_filters_parallel
[ 99%] Building Fortran object fortran/test/CMakeFiles/testhdf5_fortran.dir/fortranlib_test.F90.o
[100%] Linking Fortran executable ../../bin/testhdf5_fortran
[100%] Built target testhdf5_fortran
make: *** [Makefile:166: all] Error 2
```

The similar issue not only occurs with OpenMPI. 

When using MPICH 5.0.0 (just released), the building also has error occured, which also needs to explicitly set `-DCMAKE_Fortran_COMPILER=mpifort` to resolve: 
```
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_file_create.dir/ph5_f90_file_create.F90.o: undefined reference to symbol 'mpi_finalize_'
/usr/bin/ld: /opt/software/mpich/lib/libmpifort.so.12: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
[ 99%] Built target H5_f90_h5ex_t_vlenatt_F03
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_file_create.dir/build.make:104: bin/H5_f90_ph5_f90_file_create] Error 1
make[1]: *** [CMakeFiles/Makefile2:16427: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_file_create.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 99%] Built target H5_f90_h5ex_t_stringC_F03
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_hyperslab_by_col
[ 99%] Linking Fortran executable ../../../bin/H5_f90_exlite
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/ph5_f90_hyperslab_by_row.F90.o: undefined reference to symbol 'mpi_finalize_'
/usr/bin/ld: /opt/software/mpich/lib/libmpifort.so.12: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/build.make:104: bin/H5_f90_ph5_f90_hyperslab_by_row] Error 1
make[1]: *** [CMakeFiles/Makefile2:16458: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/all] Error 2
[ 99%] Built target H5_f90_h5ex_t_vlstring
[ 99%] Built target H5_f90_h5ex_g_visit
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/ph5_f90_hyperslab_by_col.F90.o: undefined reference to symbol 'mpi_finalize_'
/usr/bin/ld: /opt/software/mpich/lib/libmpifort.so.12: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/build.make:104: bin/H5_f90_ph5_f90_hyperslab_by_col] Error 1
make[1]: *** [CMakeFiles/Makefile2:16489: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_col.dir/all] Error 2
[ 99%] Built target H5_f90_exlite
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_dataset
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ex_ds1
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_hyperslab_by_chunk
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_hyperslab_by_pattern
/usr/bin/ld: /usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_dataset.dir/ph5_f90_dataset.F90.oCMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/ph5_f90_hyperslab_by_chunk.F90.o: undefined reference to symbol ': undefined reference to symbol 'mpi_finalize_mpi_finalize_''

/usr/bin/ld/usr/bin/ld: : /opt/software/mpich/lib/libmpifort.so.12/opt/software/mpich/lib/libmpifort.so.12: error adding symbols: : error adding symbols: DSO missing from command lineDSO missing from command line

collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/build.make:104: bin/H5_f90_ph5_f90_dataset] Error 1
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/build.make:104: bin/H5_f90_ph5_f90_hyperslab_by_chunk] Error 1
make[1]: *** [CMakeFiles/Makefile2:16396: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:16551: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_chunk.dir/all] Error 2
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/ph5_f90_hyperslab_by_pattern.F90.o: undefined reference to symbol 'mpi_finalize_'
/usr/bin/ld: /opt/software/mpich/lib/libmpifort.so.12: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/build.make:104: bin/H5_f90_ph5_f90_hyperslab_by_pattern] Error 1
make[1]: *** [CMakeFiles/Makefile2:16520: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_pattern.dir/all] Error 2
[ 99%] Built target H5_f90_ex_ds1
[ 99%] Linking Fortran executable ../../../bin/H5_f90_ph5_f90_filtered_writes_no_sel
/usr/bin/ld: CMakeFiles/H5_f90_ph5_f90_filtered_writes_no_sel.dir/ph5_f90_filtered_writes_no_sel.F90.o: undefined reference to symbol 'mpi_finalize_'
/usr/bin/ld: /opt/software/mpich/lib/libmpifort.so.12: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_filtered_writes_no_sel.dir/build.make:104: bin/H5_f90_ph5_f90_filtered_writes_no_sel] Error 1
make[1]: *** [CMakeFiles/Makefile2:16582: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_filtered_writes_no_sel.dir/all] Error 2
[ 99%] Building Fortran object fortran/test/CMakeFiles/testhdf5_fortran.dir/fortranlib_test.F90.o
[100%] Linking Fortran executable ../../bin/testhdf5_fortran
[100%] Built target testhdf5_fortran
[100%] Built target h5dumpgentest
make: *** [Makefile:166: all] Error 2
```

And that's the condition when building with MPI library of Intel oneAPI HPC Toolkit v2025.3.1, similarly needing `-DCMAKE_Fortran_COMPILER=mpiifx`:
```
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_dataset.F90(4): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [MPI]
     USE MPI
---------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_dataset.F90(28): error #6683: A kind type parameter must be a compile-time constant.   [MPI_INTEGER_KIND]
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpierror       ! MPI error flag
------------------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_dataset.F90(29): error #6683: A kind type parameter must be a compile-time constant.   [MPI_INTEGER_KIND]
     INTEGER(KIND=MPI_INTEGER_KIND) :: comm, info
------------------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_dataset.F90(30): error #6683: A kind type parameter must be a compile-time constant.   [MPI_INTEGER_KIND]
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_size, mpi_rank
------------------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_dataset.F90(32): error #6404: This name does not have a type, and must have an explicit type.   [MPI_COMM_WORLD]
     comm = MPI_COMM_WORLD
------------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_dataset.F90(33): error #6404: This name does not have a type, and must have an explicit type.   [MPI_INFO_NULL]
     info = MPI_INFO_NULL
------------^
compilation aborted for /root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_dataset.F90 (code 1)
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/build.make:75: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/ph5_f90_dataset.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16396: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_dataset.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_row.F90(7): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [MPI]
     USE MPI
---------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_row.F90(38): error #6683: A kind type parameter must be a compile-time constant.   [MPI_INTEGER_KIND]
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpierror       ! MPI error flag
------------------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_row.F90(39): error #6683: A kind type parameter must be a compile-time constant.   [MPI_INTEGER_KIND]
     INTEGER(KIND=MPI_INTEGER_KIND) :: comm, info
------------------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_row.F90(40): error #6683: A kind type parameter must be a compile-time constant.   [MPI_INTEGER_KIND]
     INTEGER(KIND=MPI_INTEGER_KIND) :: mpi_size, mpi_rank
------------------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_row.F90(42): error #6404: This name does not have a type, and must have an explicit type.   [MPI_COMM_WORLD]
     comm = MPI_COMM_WORLD
------------^
/root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_row.F90(43): error #6404: This name does not have a type, and must have an explicit type.   [MPI_INFO_NULL]
     info = MPI_INFO_NULL
------------^
compilation aborted for /root/Downloads/hdf5-develop/HDF5Examples/FORTRAN/H5PAR/ph5_f90_hyperslab_by_row.F90 (code 1)
make[2]: *** [HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/build.make:75: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/ph5_f90_hyperslab_by_row.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16458: HDF5Examples/FORTRAN/H5PAR/CMakeFiles/H5_f90_ph5_f90_hyperslab_by_row.dir/all] Error 2
```

--- a/HDF5Examples/CMakeLists.txt
+++ b/HDF5Examples/CMakeLists.txt
@@ -143,10 +143,12 @@ if (${H5_LIBVER_DIR} GREATER 16)
   if (EXISTS "${H5EXAMPLES_SOURCE_DIR}/FORTRAN" AND IS_DIRECTORY "${H5EXAMPLES_SOURCE_DIR}/FORTRAN")
     option (H5EXAMPLE_BUILD_FORTRAN "Build examples FORTRAN support" OFF)
     if (H5EXAMPLE_BUILD_FORTRAN AND HDF5_PROVIDES_FORTRAN)
+      enable_language(Fortran)
       set (H5EXAMPLE_LINK_Fortran_LIBS ${H5EXAMPLE_HDF5_LINK_LIBS})
 
       # Parallel IO usage requires MPI to be Linked and Included
       if (H5_HAVE_PARALLEL)
+        find_package(MPI REQUIRED COMPONENTS Fortran)
         set (H5EXAMPLE_LINK_Fortran_LIBS ${H5EXAMPLE_LINK_Fortran_LIBS} ${MPI_Fortran_LIBRARIES})
         if (MPI_Fortran_LINK_FLAGS)
           set (CMAKE_Fortran_EXE_LINKER_FLAGS "${MPI_Fortran_LINK_FLAGS} ${CMAKE_EXE_LINKER_FLAGS}")

